### PR TITLE
Fix GitHub Actions run failures

### LINKED ISSUE #1277: GitHub Actions are failing
The following errors are displayed when running GitHub Actions:

```
+ sudo apt-get install black gettext flake8 pylint python-is-python3 python-pytest python-setuptools-scm python3-pytest python3-pytest-flake8
Reading package lists...
Building dependency tree...
Reading state information...
Package python-pytest is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'python-pytest' has no installation candidate
E: Unable to locate package python-setuptools-scm
```

### PR DESCRIPTION
**Pull request by kurtmckee**
GitHub has updated ubuntu-latest to use Ubuntu 22.04. This release affects CI because some apt packages are no longer available for install, and a new pylint version is reporting additional lint warnings. However, while reviewing CI output, additional issues were identified and fixed.

This PR introduces these changes:

* Resolve `apt install` of packages that are renamed or unavailable on Ubuntu Jammy
* Resolve or disable pylint warnings that are showing up with newer pylint versions
* Re-enable and resolve several pylint warnings that were previously disabled
* Update GitHub Action versions to resolve multiple deprecation warnings
* Resolve pip deprecation warnings by installing wheel prior to installing Python dependencies
* Resolve severe CI artifact upload performance problems by zipping build directories
  (CI runs now take 6 minutes instead of 22 minutes)

Fixes #1277

> ## NOTE
>
> I recommend reviewing this PR commit-by-commit. All changes are self-contained and logically grouped, and reviewing individual commits might make it easier to review.

----

Previous successful CI runs showed these version numbers:

```
python -B -m pylint --version
pylint 2.4.4
astroid 2.3.3
Python 3.8.10 (default, Jun 22 2022, 20:18:18) 
```

CI is now failing with these newer versions:

```
python -B -m pylint --version
pylint 2.12.2
astroid 2.9.3
Python 3.10.6 (main, Nov  2 2022, 18:53:38) [GCC 11.3.0]
```

In addition, on my local box I'm seeing these versions, which output more new errors:

```
python -B -m pylint --version
pylint 2.15.8
astroid 2.12.13
Python 3.11.1 (main, Dec 12 2022, 18:09:30) [GCC 11.3.0]
```

### CODE REVIEW

**Comment by davvid** [comment]
Thanks Kurt, this looks great. That's a very nice commit series. I'll plan to merge this later tonight or tomorrow in case you have any other updates you'd like to include before I merge. Much appreciated!

File: cola/version.py [hunk 1]
```diff
@@ -16,9 +20,6 @@
     srcdir = os.path.dirname(os.path.dirname(__file__))
     sys.path.insert(1, srcdir)
 
-from .git import STDOUT  # noqa
-from .decorators import memoize  # noqa
-from ._version import VERSION  # noqa
```
Comment (davvid) [inline cola/version.py:21]: Makes sense. A while back in the early python2 days we deferred these imports and it allowed us to run `python cola/version.py`as a standalone script from the source tree.

We also had additional code that printed out the version (used by some of the build stuff at the time) but we got rid of that a while ago but had forgotten about the `__name__ == '__main__'` snippet here.

TODO note for myself: remove the `__main__` block here. This commit is good.. just a reminder to myself for later.

**Review by davvid — APPROVED** [review]

File: cola/widgets/createbranch.py [hunk 2]
```diff
@@ -264,7 +264,7 @@ def create_branch(self):
 
         if check_branch:
             msg = N_(
-                'Resetting "%(branch)s" to "%(revision)s" ' 'will lose commits.'
+                'Resetting "%(branch)s" to "%(revision)s" will lose commits.'
```
[the remark this answers is not in the document]
Comment (kurtmckee) [reply cola/widgets/createbranch.py:267]: Thanks, but the newer version of pylint did all the hard work of finding these! 😀

File: cola/version.py (hunk 1 above)
Comment (davvid) [reply cola/version.py:21]: Thanks Kurt, adding dependabot for sounds great.

**Pull request merged**